### PR TITLE
Add a dedicated namespace for chap secrets.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go [hunk 1]
```diff
@@ -96,6 +96,9 @@ type provisionerConfig struct {
 
 	// Optional: Chap Auth Enable
 	chapAuthEnabled bool
+
+	//Optional: Chap Auth Secret Namespace.
+	chapSecretNamespace string
```
Comment (wongma7) [inline gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: doc needs update

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go [hunk 2]
```diff
@@ -233,11 +236,15 @@ func (p *glusterBlockProvisioner) Provision(options controller.VolumeOptions) (*
 
 	if cfg.chapAuthEnabled && iscsiVol.User != "" && iscsiVol.AuthKey != "" {
 		nameSpace := options.PVC.Namespace
```
Comment (wongma7) [inline gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: should we default to a dedicated namespace and ditch the PVC namespace idea altogether?

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go (hunk 1 above)
Comment (humblec) [reply gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: true.. I was planning to hide it :).  any way I will update the doc in this PR.

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go (hunk 2 above)
Comment (humblec) [reply gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: hmmm.. then this parameter will become `mandatory` than `optional`. I am not sure whether pod spec has any restriction to use a PV which has reference to a secret from a different namespace. If not, we can mandate this and forget about `pvc namespace` altogther. @rootfs any thoughts ?

Comment (wongma7) [reply gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: not necessarily mandatory, could default to `kube-system` or something? but yeah, need second opinions.

**Comment by rootfs** [comment]
iscsi pv still references pod or pvc namespace to get chap secret. this is going to change after https://github.com/kubernetes/kubernetes/pull/51530

**Comment by humblec** [comment]
@rootfs @wongma7 then till the mentioned upstream PR is merged, we should not make this change here or in gluster block provisioner.  Isnt it ?

**Comment by rootfs** [comment]
yes, please wait until that PR is merged and client-go is updated.

**Comment by humblec** [comment]
Thanks @rootfs , Sure.! 
@wongma7 I am adding "DO-NOT-MERGE" flag on this PR :)

**Comment by humblec** [comment]
As the referenced ISCSI PR is merged, I am revisiting this PR. @rootfs @wongma7 Please review.

**Comment by humblec** [comment]
@rootfs can you please review ? :)

**Pull request closed without merging**